packages/manager/src/features/IAM/Users/UsersTable/UserRow.tsx

```diff
@@ -0,0 +1,99 @@
+import { useTheme } from '@mui/material/styles';
+import React from 'react';
+
+import { Avatar } from 'src/components/Avatar/Avatar';
+import { DateTimeDisplay } from 'src/components/DateTimeDisplay';
+import { Hidden } from 'src/components/Hidden';
+import { StatusIcon } from 'src/components/StatusIcon/StatusIcon';
+import { TableCell } from 'src/components/TableCell';
+import { TableRow } from 'src/components/TableRow';
+import { useProfile } from 'src/queries/profile/profile';
+import { capitalize } from 'src/utilities/capitalize';
+
+import { UsersActionMenu } from './UsersActionMenu';
+
+import type { User } from '@linode/api-v4';
+import { Link } from 'react-router-dom';
+import { Box, Chip, Stack, Typography } from '@linode/ui';
+
+interface Props {
+  onDelete: (username: string) => void;
+  user: User;
+}
+
+export const UserRow = ({ onDelete, user }: Props) => {
+  const theme = useTheme();
+
+  const { data: profile } = useProfile();
+
+  const isProxyUser = Boolean(user.user_type === 'proxy');
+
+  return (
+    <TableRow data-qa-table-row={user.username} key={user.username}>
+      <TableCell>
+        <Stack alignItems="center" direction="row" spacing={1.5}>
+          <Avatar
+            color={
+              user.username !== profile?.username
+                ? theme.palette.primary.dark
+                : undefined
+            }
+            username={user.username}
+          />
+          <Typography>
+            <Link to={`/iam/users/${user.username}/details`}>
+              {user.username}
+            </Link>
```

Comment:
Could we update this component to make this text [maskable](https://linode.github.io/manager/development-guide/02-component-structure.html#security), if the user chooses to mask their data via the Profile Setting? 

This would be consistent with the Users & Grants page and the Login History page, where we have started doing this.

![Screenshot 2024-12-09 at 6 49 33 PM](https://github.com/user-attachments/assets/6b7aa82a-07fa-4b65-9862-9ed5af2bfd7e)

It's as straight forward as wrapping the link in the `MaskableText` component:

```suggestion
            <MaskableText isToggleable text={user.username}>
              <Link to={`/iam/users/${user.username}/details`}>
                {user.username}
              </Link>
            </MaskableText>
```

Reply:
thanks @mjac0bs , didn't know that we have that feature!